File: data/src/main/java/com/github/gmetal/data/repository/datasource/MoviesDataSource.java
package com.github.gmetal.data.repository.datasource;

import com.github.gmetal.domain.model.PagedEntity;
import com.github.gmetal.domain.model.MediaInfo;
import com.github.gmetal.domain.model.MovieMediaDetail;
import com.github.gmetal.lib.Notifiable;

public interface MoviesDataSource {

    void getLatestMovies(final int pageNumber,
                         final Notifiable<PagedEntity<MediaInfo>, Throwable> notifiable);

    void getById(final String id, final Notifiable<MovieMediaDetail, Throwable> notifiable);
}

File: domain/src/main/java/com/github/gmetal/domain/repository/MoviesDataSource.kt
package com.github.gmetal.domain.repository

import com.github.gmetal.domain.model.PagedEntity
import com.github.gmetal.domain.model.MediaInfo
import com.github.gmetal.domain.model.MovieMediaDetail
import com.github.gmetal.lib.Notifiable

interface MoviesDataSource {

    fun getLatestMovies(pageNumber: Int,
                        notifiable: Notifiable<PagedEntity<MediaInfo>, Throwable>)

    fun getById(id: String, notifiable: Notifiable<MovieMediaDetail, Throwable>)
}
